src/frontend/planner_test/tests/testdata/output/project_set.yaml

```diff
@@ -166,3 +166,28 @@
           └─StreamProject { exprs: [Unnest($0), t._row_id, projected_row_id] }
             └─StreamProjectSet { select_list: [Unnest($0), $1] }
               └─StreamTableScan { table: t, columns: [t.x, t._row_id], pk: [t._row_id], dist: UpstreamHashShard(t._row_id) }
+- name: issue-10080
+  sql: |
+    with cte as (SELECT 1 as v1, unnest(array[1,2,3,4,5]) AS v2) select v1 from cte;
+  batch_plan: |
+    BatchProject { exprs: [1:Int32] }
+    └─BatchProjectSet { select_list: [1:Int32, Unnest(ARRAY[1, 2, 3, 4, 5]:List(Int32))] }
```

Comment:
The `1` in the ProjectSet seems sub-optimal. How to fix it? 🤪

```
LogicalProject { exprs: [1:Int32] }
 └─LogicalShare { id = 3 }
   └─LogicalProject { exprs: [1:Int32, Unnest(Array(1:Int32, 2:Int32, 3:Int32, 4:Int32, 5:Int32))] }
     └─LogicalProjectSet { select_list: [1:Int32, Unnest(Array(1:Int32, 2:Int32, 3:Int32, 4:Int32, 5:Int32))] }
       └─LogicalValues { rows: [[]], schema: Schema { fields: [] } }
 
 DAG To Tree:
 
 apply DagToTreeRule 1 time(s)
 
 LogicalProject { exprs: [1:Int32] }
 └─LogicalProject { exprs: [1:Int32, Unnest(Array(1:Int32, 2:Int32, 3:Int32, 4:Int32, 5:Int32))] }
   └─LogicalProjectSet { select_list: [1:Int32, Unnest(Array(1:Int32, 2:Int32, 3:Int32, 4:Int32, 5:Int32))] }
     └─LogicalValues { rows: [[]], schema: Schema { fields: [] } }
 
 Predicate Push Down:
 
 LogicalProject { exprs: [1:Int32] }
 └─LogicalProject { exprs: [1:Int32, Unnest(Array(1:Int32, 2:Int32, 3:Int32, 4:Int32, 5:Int32))] }
   └─LogicalProjectSet { select_list: [1:Int32, Unnest(Array(1:Int32, 2:Int32, 3:Int32, 4:Int32, 5:Int32))] }
     └─LogicalValues { rows: [[]], schema: Schema { fields: [] } }
 
 Predicate Push Down:
 
 LogicalProject { exprs: [1:Int32] }
 └─LogicalProject { exprs: [1:Int32, Unnest(Array(1:Int32, 2:Int32, 3:Int32, 4:Int32, 5:Int32))] }
   └─LogicalProjectSet { select_list: [1:Int32, Unnest(Array(1:Int32, 2:Int32, 3:Int32, 4:Int32, 5:Int32))] }
     └─LogicalValues { rows: [[]], schema: Schema { fields: [] } }

Prune Columns: 😄😄😄
 
 LogicalProject { exprs: [1:Int32] }
 └─LogicalProject { exprs: [1:Int32] }
   └─LogicalProject { exprs: [1:Int32] }
     └─LogicalProjectSet { select_list: [1:Int32, Unnest(Array(1:Int32, 2:Int32, 3:Int32, 4:Int32, 5:Int32))] }
       └─LogicalValues { rows: [[]], schema: Schema { fields: [] } }
 
```

Reply:
Maybe we can add a `output_indices` to `LogicalProjectSet`  operator?

Reply:
I am not sure why, does the LogicalProjectSet::prune_col should be called with empty requierd columns set?